devito/types/relational.py

```diff
@@ -90,6 +94,10 @@ def subdomain(self):
     def negated(self):
         return ops.get(self.func)(*self.args)
 
+    @property
+    def reversed(self):
```

Comment:
All these properties: `subdomain`, `negated`, `reversed` are **exactly the same** in all relations. 

They should rather be moved to a `RelationalMixin` class

Reply:
if you find yourself copy-pasting code, there's a very high chance you're making a mistake

Reply:
Not sure if this is easy (?). We use the self.func to build the negated class? I need to look better...